test/infrastructure/inmemory/internal/cloud/runtime/cache/client.go

```diff
@@ -117,6 +119,15 @@ func (c *cache) List(resourceGroup string, list client.ObjectList, opts ...clien
 				}
 			}
 
+			// TODO: This only matches the nodeName field for pods. No other fieldSelectors are implemented.
+			if pod, ok := obj.(*corev1.Pod); ok {
+				if listOpts.FieldSelector != nil && !listOpts.FieldSelector.Empty() {
+					if !listOpts.FieldSelector.Matches(fields.Set{"spec.nodeName": pod.Spec.NodeName}) {
```

Comment:
Is there any way to check that the "incoming" field selector is the one we think it is?

(maybe I'm misreading this entire statement)

Reply:
I'd like to take this as a follow-up for now. I'm not sure what else this might break currently.

Reply:
Sounds good

Looks like I basically added the same comment twice :joy: